onnxruntime/core/framework/utils.cc

```diff
@@ -93,6 +93,16 @@ void DefaultFree(void* p) {
 AllocatorPtr GetAllocator(const SessionState& session_state, const OrtMemoryInfo& memory_info) {
   return session_state.GetExecutionProviders().GetAllocator(memory_info);
 }
+    
+void* GetProviderRunOptions(const unordered_map<string, void*>& provider_run_options, const string& provider) {
+  void* run_options = nullptr;
```

Comment:
This feels like something that should be internal to an EP. Given the RunOptions, it should be able to read whatever it knows about directly from RunOptions.

Reply:
we would want to one XP to access the run options only belonging to it, so we obtain is out of the run options here and pass to the op_kernel_context_internal.